crates/compiler/parse/src/ast.rs

```diff
@@ -634,6 +634,16 @@ pub fn is_expr_suffixed(expr: &Expr) -> bool {
     }
 }
 
+pub fn is_top_level_expr_suffixed(expr: &Expr) -> bool {
+    match expr {
+        Expr::TaskAwaitBang(..) => true,
+        Expr::ParensAround(sub_loc_expr) => is_top_level_expr_suffixed(sub_loc_expr),
+        Expr::SpaceBefore(a, _) => is_top_level_expr_suffixed(a),
+        Expr::SpaceAfter(a, _) => is_top_level_expr_suffixed(a),
+        _ => false,
+    }
+}
```

Comment:
I think it would be helpful to include some doc comments here explaining what a top level expression means in this context. Maybe add an example